src/Symfony/Component/Translation/DataCollectorTranslator.php

```diff
@@ -82,6 +82,15 @@ public function getFallbackLocales(): array
         return [];
     }
 
+    public function getGlobalParameters(): array
+    {
+        if ($this->translator instanceof Translator || method_exists($this->translator, 'getGlobalParameters')) {
```

Comment:
is the method_exists check needed?

Reply:
One could ship their own `TranslatorInterface` implementation having global parameters support too. But maybe having a dedicated interface might be a better solution.

Reply:
this is `instanceof Translator` so the method is going to be always here to me, no?

Reply:
but `$this->translator` doesn't necessarily have to be an instance of `Translator` 🤔

Reply:
Oh, NVM, I read the `||` as `&&` :facepalm: